main.py

```diff
@@ -1,157 +1,215 @@
 
+
     def fall(self):
-        self.y+=7 #change this value if necessary
+        self.y += 7 #Change the value if necessary
+
 
     def draw(self, surface):
         try:
             surface.blit(self.image, (self.x, self.y))
         except AttributeError:
             pass
 
+
 class BlueCoin(Coin): #bonus coin
     def __init__(self):
         Coin.__init__(self)
         self.image = bluecoin
 
-class Bomb(Coin): #bomb
+
+class Bomb(Coin):
     def __init__(self):
         Coin.__init__(self)
         self.image = bomb
 
-def CoinGame():
+
+def coinGame(): #Renamed from CoinGame to coinGame
     result = 0
     i = 1
-    coinlist=[]
+    coinlist = []
     gameclock = pygame.time.Clock()
     timer = 0
     cart = Cart()
 
-    #set up texts
-    time_text = basicFont.render('TIMER:',True, BLACK, WHITE)
+    #Set up texts
+    time_text = basicFont.render('TIMER:', True, BLACK, WHITE)
     textbox = time_text.get_rect(center=(900, 170))
-    points_text = basicFont.render('POINTS:',True, BLACK, WHITE)
-    pointbox = points_text.get_rect(center=(100, 170))
-    display_time = basicFont.render('0',True, BLACK, WHITE)
+    point_text = basicFont.render('POINTS:', True, BLACK, WHITE)
+    pointbox = point_text.get_rect(center=(100, 170))
+    display_time = basicFont.render('0', True, BLACK, WHITE)
     timebox = display_time.get_rect(center=(900, 200))
-    score = basicFont.render(str(cart.Points),True, BLACK, WHITE)
+    score = basicFont.render(str(cart.points), True, BLACK, WHITE)
     scorebox = score.get_rect(center=(100, 200))
 
-    while timer<=30:
+    #Add a restart text
+    restart = basicFont.render('Press R to restart!', True, BLACK, WHITE)
+    restartbox = restart.get_rect(center=(512, 384))
+
+    over = False
+
+    while True: #Changed to infinite loop
+        if timer > 30 or cart.dead:
+            over = True
+        
         for event in pygame.event.get():
-            if event.type==QUIT:
+            if event.type == QUIT:
                 pygame.quit()
                 sys.exit()
-
-        cart.handle_keys()
-        windowSurface.blit(pygame.transform.scale(BG,(size)),(0, 0))
-        cart.draw(windowSurface)
-
-        #randomizing bonus coin/bomb/coin fall rate, can change this
-        if i%3==0 or i%4==0:
-            select = random.randint(1, 2)
-            if select==1:
-                C = BlueCoin()
-            if select==2:
-                C = Bomb()
-        elif i%5==0 or i%7==0 or i&11==0:
-            C = Bomb()
-        else:
-            C = Coin()
-        coinlist.append(C)
-        for B in coinlist[0:i:15]: #(use 14 or 15) this is for the rate at which objects fall, can change this
-            B.draw(windowSurface)
-            B.fall()
-            cart.collect_item(B)
-        pygame.display.flip()
-        i+=1
-
-        #update time
+            #Add a restart key
+            elif event.type == KEYUP:
+                if event.key == pygame.K_r and (cart.dead or over):

```

Comment:
Once the game ends, the timer restarts and still works even though 'R' isn't pressed. Please check.